File: core/java/src/main/java/hara/lang/protocol/IDerefTimeout.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.idereftimeout", name = "IDerefTimeout")
public interface IDerefTimeout<V> {
  @HaraMethod(value = "deref-timeout", arity = 3)
  V derefTimeout(long ms, V timeoutVal);
}

File: core/rust/src/lang/protocol/idereftimeout.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.idereftimeout", name = "IDerefTimeout")]
pub trait IDerefTimeout<V> {
    #[hara_method(value = "deref-timeout", arity = 3)]
    fn deref_timeout(&self, milliseconds: u64, timeout_value: V) -> V;
}
